### PR TITLE
Sign up

### PR DESCRIPTION
**Pull request by srwilliamg**
## Summary

- A new endpoint "users" was added
- Now users can be created using the endpoint POST /users/
- new Error types were added
- bcrypt library was added to encrypt the user's password
- Added error validation for sequelize

## Trello Card

https://trello.com/c/qxo7a2IU/35-sign-up

### CODE REVIEW

File: app/routes.js [hunk 1]
```diff
@@ -1,11 +1,13 @@
 const { requestHandler } = require('./helpers/requestHandler');
 
 const WitterController = require('./controllers/witter');
+const UsersController = require('./controllers/users');
 const { healthCheck } = require('./controllers/healthCheck');
 
 exports.init = app => {
   app.get('/health', healthCheck);
   app.get('/witter', requestHandler(WitterController.getWitter));
+  app.post('/users', requestHandler(UsersController.createUser));
```
Comment (blueJayWays) [inline app/routes.js:10]: I think `/signup ` is a better route name

File: app/controllers/users.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+const UsersServices = require('../services/users.js');
+const logger = require('../../app/logger/index');
+
+const WOLOX_EMAIL_DOMAIN = '@wolox';
+const UsersController = module.exports;
+
+UsersController.createUser = async (req, res) => {
+  const { body, body: { email } = {} } = req;
+
+  if (!email.includes(WOLOX_EMAIL_DOMAIN)) throw errors.badRequestError('Not valid email');
```
Comment (blueJayWays) [inline app/controllers/users.js]: This mail validation must be moved to a middleware.

File: app/controllers/users.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
```
Comment (blueJayWays) [inline app/controllers/users.js:15]: Also, you did the name and password validations at the model-level and database-level, but you missed doing the validation at the endpoint-level. As is described in the card. You could create a new middleware for this.

**Review by blueJayWays — CHANGES_REQUESTED** [review]

File: app/routes.js (hunk 1 above)
Comment (srwilliamg) [reply app/routes.js:10]: In the card the endpoint is POST /users

File: app/controllers/users.js (hunk 2 above)
Comment (srwilliamg) [reply app/controllers/users.js]: Middlewares are added in next PRs, i added extra validations

File: app/controllers/users.js (hunk 3 above)
Comment (srwilliamg) [reply app/controllers/users.js:15]: SAB

**Pull request merged**